src/util-var.h

```diff
@@ -46,7 +47,7 @@ enum VarTypes {
 };
 
 typedef struct GenericVar_ {
-    uint8_t type;
+    uint8_t type; /**< variable type, uses detection sm_type */
```

Comment:
Should we not precise that `sm_type` is u16 but the order of keywords makes it sure that we fit into an u8 ?